modules/apps/collaboration/message-boards/message-boards-editor-configuration/src/main/java/com/liferay/message/boards/editor/configuration/internal/MBEditorOptionsContributor.java

```diff
@@ -51,11 +53,18 @@ public void populateEditorOptions(
 		RequestBackedPortletURLFactory requestBackedPortletURLFactory) {
 
 		PortletURL portletURL = requestBackedPortletURLFactory.createActionURL(
-			PortletKeys.DOCUMENT_LIBRARY);
+			PortletKeys.MESSAGE_BOARDS);
 
 		portletURL.setParameter(
-			ActionRequest.ACTION_NAME, "/document_library/upload_file_entry");
-		portletURL.setParameter("folderId", "0");
+			ActionRequest.ACTION_NAME, "/message_boards/upload_temp_image");
+
+		MBMessage message = (MBMessage)themeDisplay.getRequest().getAttribute(
+			WebKeys.MESSAGE_BOARDS_MESSAGE);
+
+		long categoryId = MBUtil.getCategoryId(
+			themeDisplay.getRequest(), message);
```

Comment:
@4lejandrito This method is not returning the correct _categoryId_, it always returns "0". But is the same method used in _edit_message.jsp_ file. Maybe because of the "request" parameter?